system_metrics_collector/test/system_metrics_collector/test_linux_cpu_measurement.cpp

```diff
@@ -14,36 +14,84 @@
 
 #include <gtest/gtest.h>
 
+#include <array>
 #include <chrono>
+#include <functional>
 #include <iostream>
-#include <string>
 #include <memory>
 #include <mutex>
+#include <string>
+
+#include "metrics_statistics_msgs/msg/metrics_message.hpp"
 
 #include "../../src/system_metrics_collector/linux_cpu_measurement_node.hpp"
 #include "../../src/system_metrics_collector/proc_cpu_data.hpp"
 
+using metrics_statistics_msgs::msg::MetricsMessage;
+using metrics_statistics_msgs::msg::StatisticDataPoint;
+using metrics_statistics_msgs::msg::StatisticDataType;
+using moving_average_statistics::StatisticData;
+using moving_average_statistics::STATISTICS_DATA_TYPES;
+using system_metrics_collector::processStatCpuLine;
+
 namespace
 {
-constexpr const char proc_sample_1[] =
-  "cpu  22451232 118653 7348045 934943300 5378119 0 419114 0 0 0\n";
-constexpr const char proc_sample_2[] =
-  "cpu  22451360 118653 7348080 934949227 5378120 0 419117 0 0 0\n";
-constexpr const std::chrono::milliseconds TEST_PERIOD =
+
+constexpr const char TEST_NODE_NAME[] = "test_measure_linux_cpu";
+constexpr const char TEST_TOPIC[] = "test_cpu_measure_topic";
+
+constexpr const std::chrono::milliseconds TEST_LENGTH =
+  std::chrono::milliseconds(250);
+constexpr const std::chrono::milliseconds MEASURE_PERIOD =
   std::chrono::milliseconds(50);
-constexpr const double CPU_ACTIVE_PERCENTAGE = 2.8002699055330633;
+constexpr const std::chrono::milliseconds PUBLISH_PERIOD =
+  std::chrono::milliseconds(80);
+
+constexpr const std::array<const char *, 10> proc_samples = {
+  "cpu 22451232 118653 7348045 934943300 5378119 0 419114 0 0 0\n",
+  "cpu 22451360 118653 7348080 934949227 5378120 0 419117 0 0 0\n",
+  "cpu 24343452 61856 6484430 10645595 58695 0 683052 0 0 0\n",
+  "cpu 6051294 43322 1611333 9021635 47400 0 177494 0 0 0\n",
+  "cpu 6092443 6217 1623536 535731 4143 0 232286 0 0 0\n",
+  "cpu 6097071 6498 1612044 544445 3484 0 135942 0 0 0\n",
+  "cpu 6102643 5818 1637516 543782 3666 0 137329 0 0 0\n",
+  "cpu 24513632 62372 6527524 11205004 62394 0 687176 0 0 0\n",
+  "cpu 6093617 43419 1621953 9161570 48047 0 178792 0 0 0\n",
+  "cpu 6134250 6371 1634411 675446 5285 0 233478 0 0 0\n"
+};
+
+double computeCpuActivePercentage(const char * data1, const char * data2)
+{
+  auto parsed_data1 = processStatCpuLine(data1);
+  auto parsed_data2 = processStatCpuLine(data2);
+  return computeCpuActivePercentage(parsed_data1, parsed_data2);
+}
+
+constexpr const double CPU_ACTIVE_PROC_SAMPLE_0_1 = 2.7239908106334099;
+
+void StatisticDataToStatisticDataPoints(const StatisticData & src, StatisticDataPoint * dst)
+{
+  dst[StatisticDataType::STATISTICS_DATA_TYPE_AVERAGE - 1].data = src.average;
```

Comment:
once it's a STL vector, we can use back() here instead

Reply:
With it now being a `std::unordered_map`, it is no longer relevant.